### ISSUE TITLE
Broken RegExp for pattern with noext option

### ISSUE THREAD

**Issue by mrmlnc** [micromatch/micromatch#116]
## Please describe the **minimum necessary steps** to reproduce this issue:

Just run the following code:

```js
const micromatch = require('micromatch');

const re = micromatch.makeRe('./test(dir)/foo.txt', { noext: true });

console.dir(re, { colors: true });
```

## What is happening (but shouldn't):

In the `re` variable we have a the following RegExp:

```js
/^(?:(?:(?:\.(?:\/|\\))(?=.))?test\(\dir\)\/foo\.txt)$/
// ----------------------------------^
```

As we see, `\d` in this case will match digits instead of the `d` character.

## What should be happening instead?

The `\d` symbol should be `d`.

**Comment by mrmlnc** [comment]
The problem is here: 

https://github.com/micromatch/micromatch/blob/master/lib/compilers.js#L53

We shouldn't add `\\` to text node.

**Comment by jonschlinkert** [comment]
good catch! thanks for the issue.

**Comment by mrmlnc** [comment]
Should be fixed by `3.1.6`.

**Comment by jonschlinkert** [comment]
awesome thanks! Published and released as 3.1.6. I appreciate the issue and PR!
